python/samples/agentchat_graphrag/app.py

```diff
@@ -0,0 +1,77 @@
+import argparse
+import asyncio
+import json
+import logging
+from typing import Any, Dict
+from autogen_agentchat.messages import TextMessage
+from autogen_ext.tools.graphrag import (
+    GlobalSearchTool,
+    LocalSearchTool,
+)
+from autogen_agentchat.agents import AssistantAgent
+from autogen_agentchat.conditions import TextMentionTermination, MaxMessageTermination
+from autogen_agentchat.teams import RoundRobinGroupChat
+from autogen_core.models import ChatCompletionClient
+
+
+async def main(model_config: Dict[str, Any]) -> None:
+    # Initialize the model client from config
+    model_client = ChatCompletionClient.load_component(model_config)
+
+    # Set up global search tool
+    global_tool = GlobalSearchTool.from_settings(
+        settings_path="./settings.yaml"
+    )
+
+    local_tool = LocalSearchTool.from_settings(
+        settings_path="./settings.yaml"
+    )
+
+    # Create assistant agent with both search tools
+    assistant_agent = AssistantAgent(
+        name="search_assistant",
+        tools=[global_tool, local_tool],
+        model_client=model_client,
+        system_message=(
+            "You are a tool selector AI assistant using the GraphRAG framework. "
+            "Your primary task is to determine the appropriate search tool to call based on the user's query. "
+            "For specific, detailed information about particular entities or relationships, call the 'local_search' function. "
+            "For broader, abstract questions requiring a comprehensive understanding of the dataset, call the 'global_search' function. "
+            "Do not attempt to answer the query directly; focus solely on selecting and calling the correct function."
+        )
+    )
+
+    # Set up the team
+    termination = TextMentionTermination("TERMINATE") | MaxMessageTermination(10)
+    team = RoundRobinGroupChat(
```

Comment:
Instead of using `RoundRobinGroupChat`, just call `run_stream` directly on the assistant agent. This is to avoid underministic behaviors when the agent doesn't produce "TERMINATE" and the team keeps going.

Reply:
Use `autogen_agentchat.ui.Console` instead of rolling your own rendering to produce all output.